### PR TITLE
fix handle major version changes correctly

### PR DESCRIPTION
**Pull request by funkenstrahlen**
This is my suggestion for a fix. I wanted to create some test cases for this but could not find any unit testing targets. As I do not want to change your project structure too much with this simple fix I did not add all that too. Please think about adding testing!

This is related to #22

### CODE REVIEW

**Comment by BalestraPatrick** [comment]
Hey @funkenstrahlen! There is a unit tests target, you can find it directly here: https://github.com/BalestraPatrick/WhatsNew/blob/master/Example/Tests/WhatsNewTests.swift

**Comment by funkenstrahlen** [comment]
Ah Great! I will have a look at the tests and update my PR! Did not have a look into the example folder.

**Comment by funkenstrahlen** [comment]
Your test setup is not very flexible and kinda weird because the current version number for the app *always* is 1.0 as it is loaded from the application bundle. I will look for a way to improve that.

File: WhatsNew/Items/WhatsNew.swift [hunk 1]
```diff
@@ -9,31 +9,38 @@ import Foundation
...
         switch option {
         case .debug: return true
         case .never: return false
-        case .majorVersion: return didChangeMajorVersion(previous: previousAppVersion, current: appVersion)
+        case .majorVersion: return didUpdate && isMajorVersion(version: currentAppVersion)
         case .always: return didUpdate
         }
     }
-
-    private static func didChangeMajorVersion(previous: String?, current: String?) -> Bool {
```
Comment (BalestraPatrick) [inline WhatsNew/Items/WhatsNew.swift:34]: I don't see the problem with my implementation of `didChangeMajorVersion` in this case?

File: WhatsNew/Items/WhatsNew.swift [hunk 2]
```diff
@@ -9,31 +9,38 @@ import Foundation
 
 public struct WhatsNew {
     static let bundle = Bundle(for: WhatsNewViewController.self)
-    static let appVersion = Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String
-    static let userDefaultsKey = "LatestAppVersionPresented"
+    public static let appVersion = Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String
```
Comment (BalestraPatrick) [inline WhatsNew/Items/WhatsNew.swift]: Can you rename this variable to `currentVersion` and move it before all the `internal` declarations so it's easier to see it's `public`?

File: Example/Tests/WhatsNewTests.swift [hunk 3]
```diff
@@ -17,37 +17,52 @@ class WhatsNewTests: XCTestCase {
         UserDefaults.standard.removePersistentDomain(forName: Bundle.main.bundleIdentifier!)
     }
     
+    func testShouldNotPresent_WhenCurrentVersionIsUndefined() {
+        XCTAssertFalse(WhatsNew.shouldPresent(with: .always, currentVersion: nil))
+    }
+    
+    
+
```
Comment (BalestraPatrick) [inline Example/Tests/WhatsNewTests.swift]: Can you please remove these spacing between the methods and only leave one empty line between each test? 👍

**Review by BalestraPatrick — CHANGES_REQUESTED** [review]

File: WhatsNew/Items/WhatsNew.swift (hunk 2 above)
Comment (funkenstrahlen) [reply WhatsNew/Items/WhatsNew.swift]: of course!

File: Example/Tests/WhatsNewTests.swift (hunk 3 above)
Comment (funkenstrahlen) [reply Example/Tests/WhatsNewTests.swift]: yes

File: WhatsNew/Items/WhatsNew.swift (hunk 1 above)
Comment (funkenstrahlen) [reply WhatsNew/Items/WhatsNew.swift:34]: For `previous = nil` your implementation did not present the whats new controller when presentation option was set to `.major`.

Comment (funkenstrahlen) [reply WhatsNew/Items/WhatsNew.swift:34]: This is why I changed the implementation to a better understandable way: I check for `didUpdate && isMajorVersion`.

**Comment by funkenstrahlen** [comment]
@BalestraPatrick Can you have another look?

**Review by BalestraPatrick — APPROVED** [review]

**Comment by funkenstrahlen** [comment]
@BalestraPatrick Just to be sure:

Version change from 1.9 to 2.1 should not present on presentation option `.major`?

This is a difficult case because the information passed to whats new controller for the 2.1 release might not include information for the 2.0 major release. So I think it is correct to not present in this case.

**Comment by BalestraPatrick** [comment]
@funkenstrahlen Yes, it should present. Major is the first number in the version, so if it changes from 1 to 2 it should present. So if either there is no previous version stored, or if the previous version is < than the current one, we should present if presentation option is `.major`.

**Comment by funkenstrahlen** [comment]
Ok good to know. In this case my code does not work correct and the test cases need to be adjusted.

**Pull request merged**